addons/mail/controllers/thread.py

```diff
@@ -64,7 +64,7 @@ def mail_thread_messages(self, thread_model, thread_id, fetch_params=None):
         res = request.env["mail.message"]._message_fetch(domain=None, thread=thread, **(fetch_params or {}))
         messages = res.pop("messages")
         if not request.env.user._is_public():
-            messages.set_message_done()
+            messages.filtered_domain([("message_type", "!=", "user_notification")]).set_message_done()
```

Comment:
Why? We want them marked as read, it's the whole point of the change.

Reply:
Check with aju but I think we want to either:
- display them forever
- display them the very first time while they are unread, and then not display them again. 

So basically they need to be displayed at least once, and they need to be marked as read immediately when they are shown.